components/widgets/charts/VegaChartTooltip.js

```diff
@@ -25,60 +25,59 @@ class VegaChartTooltip extends React.Component {
     return x.value;
   }
 
-  getParsedY() {
-    const { item } = this.props;
-    if (!item) return null;
-
-    const { y } = item;
-    if (!y) return null;
-
-    if (y.format && y.type === 'number') {
-      return format(y.format)(y.value);
-    } else if (y.type === 'date') {
-      const date = new Date(y.value);
+  getParsedValue(field) { // eslint-disable-line class-methods-use-this
+    if (field.format && field.type === 'number') {
+      return format(field.format)(field.value);
+    } else if (field.type === 'date') {
+      const date = new Date(field.value);
       // NOTE: it's important to have a default format for
-      // the manually-created widgets, otherwise if y.format
+      // the manually-created widgets, otherwise if field.format
       // is not defined, time.format will return a date
       // object and the app will crash in dev environment
       // and the tooltip won't show in prod
-      const f = y.format || '%d %b %Y';
+      const f = field.format || '%d %b %Y';
       return time.format(f)(date);
     }
 
-    return y.value;
+    return field.value;
   }
 
-
   render() {
+    const fieldsName = Object.keys(this.props.item).filter(f => f !== 'x');
     return (
       <div className="c-chart-tooltip">
         { this.props.item.x.label && (
           <div className="labels">
-            { this.props.item.y.label && <span>{this.props.item.y.label}</span> }
+            { fieldsName.length && fieldsName.map(fieldName => (
+              this.props.item[fieldName].label
+                ? <span>{this.props.item[fieldName].label}</span>
+                : false
+            ))}
             <span>{this.props.item.x.label}</span>
           </div>
         )}
         <div className="values">
-          { this.props.item.y.value && <span>{this.getParsedY()}</span> }
+          { fieldsName.length && fieldsName.map(fieldName => (
+            this.props.item[fieldName].value
+              ? <span>{this.getParsedValue(this.props.item[fieldName])}</span>
+              : false
+          ))}
           <span>{this.getParsedX()}</span>
         </div>
       </div>
     );
   }
 }
 
-const columnType = PropTypes.shape({
-  type: PropTypes.oneOf(['number', 'string', 'date']),
-  label: PropTypes.string,
-  format: PropTypes.string,
-  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.object])
-});
+// const columnType = PropTypes.shape({
```

Comment:
If we are not using it, remove it. If you want to preserve it for future implementations don't read this....oh!! you already did it...

Reply:
Actually, I wanted to still use it but `prop-types` don't let me define the props as I wanted. I just let the text there so we remember the structure in the future.